src/Action.hs

```diff
@@ -251,9 +253,11 @@ currentUser = do
     uid <- currentUserId
     muser <- query (findUser uid)
     case muser of
-        Just user -> pure user
-        Nothing   -> logout >> throwError500 "Unknown user identitifer"
+        Just user | isActiveUser user
+            -> pure user
+        _   -> logout >> throwError500 "Unknown user identitifer"
 
+-- TODO: Check if the current user is not deleted.
```

Comment:
Race condition:

 * Somebody changes her profile
 * The admin deletes the user

I see cases when the user gets undeleted by the modification action.